webhooks/virtualmachine/validation/virtualmachine_validator.go

```diff
@@ -1154,3 +1167,46 @@ func (v validator) validateLabel(ctx *pkgctx.WebhookRequestContext, vm, oldVM *v
 
 	return allErrs
 }
+
+func (v validator) validateNetworkHostAndDomainName(
+	ctx *pkgctx.WebhookRequestContext,
+	vm *vmopv1.VirtualMachine) field.ErrorList {
+
+	var allErrs field.ErrorList
+	f := field.NewPath("spec", "network")
+
+	hn, dn := vm.Spec.Network.HostName, vm.Spec.Network.DomainName
+
+	if v := hn; v != "" {
```

Comment:
Out of curiosity, why create a copy of `hn` and not do `hn != nil`?

Reply:
So the inner logic is copy/paste-able between the two areas.